lib/client/api.go

```diff
@@ -852,36 +847,50 @@ func (tc *TeleportClient) getProxyLogin() string {
 	return proxyLogin
 }
 
+// authMethods returns a list (slice) of all SSH auth methods this client
+// can use to try to authenticate
+func (tc *TeleportClient) authMethods() []ssh.AuthMethod {
+	// return the auth methods that we were configured with
+	// plus our local key agent (i.e. methods we've added during runtime
+	// by the means of .AddKey())
+	m := append([]ssh.AuthMethod(nil), tc.Config.AuthMethods...)
```

Comment:
FWTW I started using it because it promises to be more efficient, avoiding re-allocating an empty slice